core/src/main/java/org/commonjava/maven/ext/core/impl/PreparseGroovyManipulator.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2012 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.maven.ext.core.impl;
+
+import groovy.lang.GroovyShell;
+import groovy.lang.Script;
+import org.apache.maven.repository.DefaultMirrorSelector;
+import org.commonjava.maven.ext.common.ManipulationException;
+import org.commonjava.maven.ext.common.model.Project;
+import org.commonjava.maven.ext.core.ManipulationSession;
+import org.commonjava.maven.ext.core.groovy.InvocationPoint;
+import org.commonjava.maven.ext.core.groovy.InvocationStage;
+import org.commonjava.maven.ext.core.state.GroovyState;
+import org.commonjava.maven.ext.core.state.State;
+import org.commonjava.maven.ext.io.FileIO;
+import org.commonjava.maven.ext.io.ModelIO;
+import org.commonjava.maven.ext.io.PomIO;
+import org.commonjava.maven.ext.io.resolver.GalleyAPIWrapper;
+import org.commonjava.maven.ext.io.resolver.GalleyInfrastructure;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * {@link Manipulator} implementation that can resolve a remote groovy file and execute it on executionRoot. Configuration
```

Comment:
Minor : technically not a org.commonjava.maven.ext.core.impl.Manipulator implementation :-)